dask_expr/groupby.py

```diff
@@ -321,9 +329,27 @@ def _single_agg(
                 self.dropna,
                 chunk_kwargs=chunk_kwargs,
                 aggregate_kwargs=aggregate_kwargs,
+                _slice=self._slice,
             )
         )
 
+    def __getattr__(self, key):
+        try:
+            return self[key]
+        except KeyError as e:
+            raise AttributeError(e) from e
+
+    def __getitem__(self, key):
+        g = GroupBy(
```

Comment:
We didn't feel the need to distinguish between `DataFrame`/`SeriesGroupBy` before this PR. Curious if you still think we can use the same class for both?

Reply:
This PR doesn't really change anything, since we control the result via ``slice``, but there are some methods that are only defined on ``SeriesGroupBy`` and not on ``DataFrameGroupBy``, so we _might_ need both classes later on.